modules/stochastic_tools/src/state_sim/model/actions/StateSimActionList.C

```diff
@@ -0,0 +1,52 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateSimActionList.h"
+#include "StateSimAction.h"
+#include <unordered_map>
+
+StateSimActionList::StateSimActionList(bool move_from_current)
+  : _move_from_current(move_from_current),
+    _actions()
+{
+}
+
+// StateSimActionList::StateSimActionList(StateSimAction & init_act, bool move_from_current)
+//   : _move_from_current(move_from_current),
+//     _actions()
+// {
+//   _init_act.insert(std::make_pair(init_act->id(), init_act));
+// }
+
+void
+StateSimActionList::add(StateSimAction * to_add)
+{
+  std::unordered_map<int, StateSimAction *>::const_iterator got = _actions.find(to_add->id());
+  if (got == _actions.end())
+  {
+    _actions.insert(std::make_pair(to_add->id(), to_add));
```

Comment:
Minor: `find` followed by `insert` does two lookups. You can save the result of find and use it as a hint for insertion. This may not be true for unordered map though... I haven't looked at that.